File: Competitive Programming 3/Section 1.3.3/IOI '09 - Plovdiv, Bulgaria POI/Python/main.py
t,n,solved,problemAntiScore=0,0,[],[]
def calcScore(person):
    score = 0
    for j in range(t):
        if solved[person][j]:
            score += n - problemAntiScore[j]
    return score
def main():
    global t,n,problemAntiScore
    n,t,p=map(int,input().split())
    p-=1
    problemAntiScore,solvedProblems=[0] * t,[0]*n
    for i in range(n):
        row = list(map(int,input().split()))
        solved.append(row)
        for j,number in enumerate(row):
            if number == 1:
                problemAntiScore[j]+=1
                solvedProblems[i]+=1
    philipScore,rank = calcScore(p),1
    for i in range(n):
        if i!=p:
            scoreTemp = calcScore(i)
            if scoreTemp>philipScore or (scoreTemp==philipScore and solvedProblems[i]>solvedProblems[p]) or (scoreTemp==philipScore and solvedProblems[i]==solvedProblems[p] and i<p):
                #i goes before philip
                rank+=1
    print(str(philipScore)+' '+str(rank))
main()

File: Competitive Programming 3/Section 1.3.3/IOI '09 - Plovdiv, Bulgaria POI/C++/main.cpp
#include<iostream>

#define maxV 2000

using namespace std;

int problemAntiscore[maxV],solvedProblems[maxV],n,t,p;
bool solved[maxV][maxV];

int calcScore(int person){
    int score = 0;
    for(int j = 0 ; j < t; j++)
        if(solved[person][j])
            score += n - problemAntiscore[j];
    return score;
}

int main(){
    ios_base::sync_with_stdio(0);cin.tie(0);
    cin >> n >> t >> p;
    p--;
    for(int i = 0 ; i < n ; i ++){
        for(int j = 0 ; j < t ; j++){
            cin >> solved[i][j];
            if(solved[i][j]){
                problemAntiscore[j]++;
                solvedProblems[i]++;
            }
        }
    }
    int philipScore = calcScore(p),rank=1;
    for(int i = 0 ; i < n ;i++){
        if(i!=p){
            int scoreTemp = calcScore(i);
            if(scoreTemp>philipScore||(scoreTemp==philipScore&&solvedProblems[i]>solvedProblems[p])||(scoreTemp==philipScore&&solvedProblems[i]==solvedProblems[p]&&i<p))
                //i goes before philip
                rank++;
        }
    }
    cout << philipScore << " " << rank << "\n";
    return 0;
}